internal/handlers/common/scale.go

```diff
@@ -0,0 +1,86 @@
+// Copyright 2021 FerretDB Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package common
+
+import (
+	"errors"
+	"fmt"
+	"math"
+
+	"github.com/FerretDB/FerretDB/internal/handlers/commonerrors"
+	"github.com/FerretDB/FerretDB/internal/types"
+	"github.com/FerretDB/FerretDB/internal/util/lazyerrors"
+)
+
+// GetScaleParam validates the given scale value for collStats and dbStats command.
+//
+// If the value is valid, it returns its int32 representation,
+// otherwise it returns a command error with the given command being mentioned.
+func GetScaleParam(command string, value any) (int32, error) {
```

Comment:
We should try to use `NewCommandErrorMsgWithArgument` instead of `NewCommandError` where we can. It helps us collect stats :hugs:

Reply:
Fixed